src/components/views/context_menus/DeviceContextMenu.tsx

```diff
@@ -47,21 +48,23 @@ interface IDeviceContextMenuSectionProps {
 }
 
 const DeviceContextMenuSection: React.FC<IDeviceContextMenuSectionProps> = ({ deviceKind }) => {
-    const [devices, setDevices] = useState<MediaDeviceInfo[]>([]);
-    const [selectedDevice, setSelectedDevice] = useState(MediaDeviceHandler.getDevice(deviceKind));
+    const devices = useAsyncMemo(async () =>
+        (await MediaDeviceHandler.getDevices())?.[deviceKind],
+    [deviceKind]) ?? [];
+    let [selectedDevice, setSelectedDevice] = useState(MediaDeviceHandler.getDevice(deviceKind));
 
-    useEffect(() => {
-        const getDevices = async () => {
-            return setDevices((await MediaDeviceHandler.getDevices())[deviceKind]);
-        };
-        getDevices();
-    }, [deviceKind]);
+    if (!devices.length) return null;
 
     const onDeviceClick = (deviceId: string): void => {
         MediaDeviceHandler.instance.setDevice(deviceId, deviceKind);
         setSelectedDevice(deviceId);
     };
 
+    if (!devices.some(d => d.deviceId === selectedDevice)) {
+        // Ensure that we select a default device
+        selectedDevice = devices[0].deviceId;
+    }
```

Comment:
Couldn't this potentially lead to us showing a selected device that doesn't match the reality?

Reply:
That's true… I guess a better approach would  be to ensure that the lobby sets the settings value to what it considers to be the default device